src/httpclient.c

```diff
@@ -251,11 +253,22 @@ static void start_request(h2o_httpclient_ctx_t *ctx)
         return;
     }
 
+    if (req.connect_to != NULL) {
+        req.connect_to->scheme = url_parsed->scheme;
+
+        /* rebuild url_parsed->authority with the port string, even for default ports */
+        if (h2o_url_init_with_hostport(url_parsed, &pool, url_parsed->scheme, url_parsed->host, h2o_url_get_port(url_parsed),
+                                       url_parsed->path, 0) != 0) {
+            on_error(ctx, "failed to rebuild URL: %s", req.url);
+            return;
+        }
```

Comment:
Could you please elaborate why we need this? It seems like regenerating `url_parsed` from the existing values of `url_parsed`.

Reply:
@kazuho without this change, the `CONNECT` method fails if you omit the default port:
```
./h2o-httpclient -x 127.0.0.1:8080 -m CONNECT http://www.nalramli.com/index.html
HTTP/1.1 400 Bad Request
Connection: keep-alive
Content-Length: 11
Server: h2o/2.3.0-DEV@612cbbc
content-type: text/plain; charset=utf-8

Bad Request
```
It only works if you explicitly include the port:
```
$ ./h2o-httpclient -x 127.0.0.1:8080 -m CONNECT http://www.nalramli.com:80/index.html
HTTP/1.1 200 OK
Connection: close
Server: h2o/2.3.0-DEV@612cbbc

GET / HTTP/1.1
Host: www.nalramli.com

HTTP/1.1 200 OK
Content-Length: 0
Connection: close
Location: https://www.nalramli.com/

socket closed by peer
```

I think it has to do with how the http library handles `CONNECT` requests.